### PR TITLE
[New Rule] Potential PowerShell HackTool Script by Author

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Detects known PowerShell offensive tooling author's name in PowerShell scripts. Attackers commonly use out-of-the-box offensive tools without modifying the code. This rule aim is to take advantage of that.

### CODE REVIEW

**Review by Samirbous — APPROVED** [review]

File: rules/windows/execution_posh_hacktool_authors.toml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+maturity = "production"
+updated_date = "2023/01/17"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects known PowerShell offensive tooling author's name in PowerShell scripts. Attackers commonly use out-of-the-box
+offensive tools without modifying the code. This rule aim is to take advantage of that.
```
Comment (terrancedejesus) [inline rules/windows/execution_posh_hacktool_authors.toml:13]:
```suggestion
Detects known PowerShell offensive tooling author's name in PowerShell scripts. Attackers commonly use out-of-the-box
offensive tools without modifying the code, which may still contain the author artifacts. This rule identifies common author handles found in popular PowerShell scripts used for red team exercises.
```

File: rules/windows/execution_posh_hacktool_authors.toml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+rule_id = "2553a9af-52a4-4a05-bb03-85b2a479a0a0"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Execution", "PowerShell"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.category:process and
+  powershell.file.script_block_text : (
+    "JosephBialek" or "mattifestation" or
```
Comment (terrancedejesus) [inline rules/windows/execution_posh_hacktool_authors.toml:51]: Do we need to add wildcards to these filters by chance or will these substrings be identified in the full keyword strings?

**Review by terrancedejesus — APPROVED** [review]
LGTM! Added a few suggestions.

**Review by Aegrah — APPROVED** [review]
LGTM.

File: rules/windows/execution_posh_hacktool_authors.toml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+]
+risk_score = 47
+rule_id = "2553a9af-52a4-4a05-bb03-85b2a479a0a0"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Execution", "PowerShell"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.category:process and
```
Comment (terrancedejesus) [inline rules/windows/execution_posh_hacktool_authors.toml:49]:
```suggestion
host.os.type:windows and event.category:process and
```

File: rules/windows/execution_posh_hacktool_authors.toml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+maturity = "production"
+updated_date = "2024/05/08"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects known PowerShell offensive tooling author's name in PowerShell scripts. Attackers commonly use out-of-the-box
+offensive tools without modifying the code, which may still contain the author artifacts. This rule identifies common author handles found in popular PowerShell scripts used for red team exercises.
```
Comment (Samirbous) [inline rules/windows/execution_posh_hacktool_authors.toml:13]: may need a lint

File: rules/windows/execution_posh_hacktool_authors.toml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+```
+reg add "hklm\\SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging" /v EnableScriptBlockLogging /t REG_DWORD /d 1
+```
+"""
+references = [
+    "https://github.com/atc-project/atc-data/blob/master/docs/Logging_Policies/LP_0109_windows_powershell_script_block_log.md",
+]
+risk_score = 47
+rule_id = "2553a9af-52a4-4a05-bb03-85b2a479a0a0"
+severity = "medium"
```
Comment (Samirbous) [inline rules/windows/execution_posh_hacktool_authors.toml:43]:
```suggestion
severity = "high"
```

File: rules/windows/execution_posh_hacktool_authors.toml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+Steps to implement the logging policy via registry:
+
+```
+reg add "hklm\\SOFTWARE\\Policies\\Microsoft\\Windows\\PowerShell\\ScriptBlockLogging" /v EnableScriptBlockLogging /t REG_DWORD /d 1
+```
+"""
+references = [
+    "https://github.com/atc-project/atc-data/blob/master/docs/Logging_Policies/LP_0109_windows_powershell_script_block_log.md",
+]
+risk_score = 47
```
Comment (Samirbous) [inline rules/windows/execution_posh_hacktool_authors.toml:41]:
```suggestion
risk_score = 73
```

**Review by Samirbous — APPROVED** [review]

**Review by terrancedejesus — APPROVED** [review]
Nice work. Lint, if not already, and we should be good to go.

**Pull request merged**